### PR TITLE
test: add timeout for dev entries to avoid hard navigation

### PR DESCRIPTION
**Pull request by huozhi**
x-ref: https://github.com/vercel/next.js/pull/32472#issuecomment-992984581

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: db4c60e28da79ef5b466ce18d74df20b82a39187

**test/integration/app-document-add-hmr/test/index.test.js** 
- _app/_document add HMR > should HMR when _document is added

<details>
<summary>Expand output</summary>

  ● _app/_document add HMR › should HMR when _document is added

    TIMED OUT: success

    <head><style data-next-hide-fouc="true">body{display:none}</style><noscript data-next-hide-fouc="true"><style>body{display:block}</style></noscript><meta charset="utf-8"><meta name="viewport" content="width=device-width"><meta name="next-head-count" content="2"><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/_next/static/chunks/polyfills.js?ts=1639439551163"></script><script src="/_next/static/chunks/webpack.js?ts=1639439551163" defer=""></script><script src="/_next/static/chunks/main.js?ts=1639439551163" defer=""></script><script src="/_next/static/chunks/pages/_app.js?ts=1639439551163" defer=""></script><script src="/_next/static/chunks/pages/index.js?ts=1639439551163" defer=""></script><script src="/_next/static/development/_buildManifest.js?ts=1639439551163" defer=""></script><script src="/_next/static/development/_ssgManifest.js?ts=1639439551163" defer=""></script><script src="/_next/static/development/_middlewareManifest.js?ts=1639439551163" defer=""></script><noscript id="__next_css__DO_NOT_USE__"></noscript></head><body><div id="__next" data-reactroot=""><p>custom _app</p><p>index page</p></div><script src="/_next/static/chunks/react-refresh.js?ts=1639439551163"></script><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{}},"page":"/","query":{},"buildId":"development","nextExport":true,"autoExport":true,"isFallback":false,"scriptLoader":[]}</script></body>

      479 |     if (!this.originalContent) {
      480 |       this.originalContent = content
    > 481 |     }
          |      ^
      482 |     writeFileSync(this.path, content, 'utf8')
      483 |   }
      484 |

      at Object.check (lib/next-test-utils.js:481:15)
      at Object.<anonymous> (integration/app-document-add-hmr/test/index.test.js:83:13)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| buildDuration | 21.7s | 22s | ⚠️ +329ms |
| buildDurationCached | 3.9s | 4s | ⚠️ +105ms |
| nodeModulesSize | 350 MB | 350 MB | ⚠️ +4 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.534 | 3.516 | -0.02 |
| / avg req/sec | 707.43 | 711.04 | +3.61 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.63 | 1.653 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1533.7 | 1512.34 | ⚠️ -21.36 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 30.2 kB | 30.2 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 74 kB | 74 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.39 kB | 2.39 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.73 kB | 4.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.13 kB | 2.13 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| buildDuration | 23.3s | 23.9s | ⚠️ +539ms |
| buildDurationCached | 4s | 4.1s | ⚠️ +108ms |
| nodeModulesSize | 350 MB | 350 MB | ⚠️ +4 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.528 | 3.521 | -0.01 |
| / avg req/sec | 708.6 | 710.1 | +1.5 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.646 | 1.64 | -0.01 |
| /error-in-render avg req/sec | 1518.79 | 1524.17 | +5.38 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 30.3 kB | 30.3 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 74.2 kB | 74.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.39 kB | 2.39 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 906 B | 906 B | ✓ |
| image-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
Commit: db4c60e28da79ef5b466ce18d74df20b82a39187

File: test/integration/react-streaming-and-server-components/app/next.config.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const withReact18 = require('../../react-18/test/with-react-18')
 
 module.exports = withReact18({
+  maxInactiveAge: 1000 * 60 * 60,
```
Comment (ijjk) [inline test/integration/react-streaming-and-server-components/app/next.config.js:4]:
```suggestion
  onDemandEntries: {
    // Make sure entries are not getting disposed.
    maxInactiveAge: 1000 * 60 * 60,
  },
```

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| buildDuration | 22.1s | 22.2s | ⚠️ +55ms |
| buildDurationCached | 4.1s | 4.3s | ⚠️ +200ms |
| nodeModulesSize | 350 MB | 350 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.555 | 3.564 | ⚠️ +0.01 |
| / avg req/sec | 703.23 | 701.39 | ⚠️ -1.84 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.651 | 1.666 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 1513.95 | 1500.57 | ⚠️ -13.38 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 30.2 kB | 30.2 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 74 kB | 74 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.39 kB | 2.39 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.73 kB | 4.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.13 kB | 2.13 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| buildDuration | 23.8s | 23.9s | ⚠️ +11ms |
| buildDurationCached | 4.1s | 4.1s | ⚠️ +16ms |
| nodeModulesSize | 350 MB | 350 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.585 | 3.406 | -0.18 |
| / avg req/sec | 697.39 | 734.01 | +36.62 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.657 | 1.589 | -0.07 |
| /error-in-render avg req/sec | 1508.45 | 1573.22 | +64.77 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 30.3 kB | 30.3 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 74.2 kB | 74.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.39 kB | 2.39 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 906 B | 906 B | ✓ |
| image-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
Commit: 2ea37d49f13815de0f5c33eb6f3ad442d819d464

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 2ea37d49f13815de0f5c33eb6f3ad442d819d464

**test/integration/react-streaming-and-server-components/test/index.test.js** 
- concurrentFeatures - dev > should support next/link

<details>
<summary>Expand output</summary>

  ● concurrentFeatures - dev › should support next/link

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: undefined

      259 |     )
      260 |     const dynamicRouteHTML2 = await renderViaHTTP(
    > 261 |       context.appPort,
          |                       ^
      262 |       '/routes/dynamic2'
      263 |     )
      264 |

      at Object.<anonymous> (integration/react-streaming-and-server-components/test/index.test.js:261:56)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| buildDuration | 21.5s | 21.9s | ⚠️ +459ms |
| buildDurationCached | 4.1s | 3.9s | -136ms |
| nodeModulesSize | 350 MB | 350 MB | -1 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.392 | 3.54 | ⚠️ +0.15 |
| / avg req/sec | 736.97 | 706.16 | ⚠️ -30.81 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.532 | 1.592 | ⚠️ +0.06 |
| /error-in-render avg req/sec | 1631.85 | 1570.73 | ⚠️ -61.12 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 30.2 kB | 30.2 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 74 kB | 74 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.39 kB | 2.39 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.73 kB | 4.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.13 kB | 2.13 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| index.html gzip | 532 B | 532 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| buildDuration | 22.4s | 22.8s | ⚠️ +435ms |
| buildDurationCached | 3.9s | 3.8s | -83ms |
| nodeModulesSize | 350 MB | 350 MB | -1 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.633 | 3.455 | -0.18 |
| / avg req/sec | 688.08 | 723.58 | +35.5 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.621 | 1.599 | -0.02 |
| /error-in-render avg req/sec | 1542.44 | 1563.78 | +21.34 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 30.3 kB | 30.3 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 74.2 kB | 74.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.39 kB | 2.39 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 906 B | 906 B | ✓ |
| image-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js test/ondemand-entries-timeout | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 527 B | 527 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
Commit: cae8c04a51a897bf7d733bbe83e0c606e0e80b98

**Review by timneutkens — APPROVED** [review]

**Pull request merged**